contracts/utils/TransferHelper.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity 0.8.15;
+
+import "../libraries/Constants.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract TransferHelper {
+    using SafeERC20 for IERC20;
+
+    /// @notice transfer tokens in if necessary
+    /// @param token input token
+    /// @param inputAmount input amount
+    function _transferIn(IERC20 token, uint256 inputAmount) internal returns (uint256) {
+        if (inputAmount == Constants.CONTRACT_BALANCE) {
+            inputAmount = token.balanceOf(address(this));
```

Comment:
There are a few places where `_getBalance` can be used. I don't really care which way we go, but it would be nice to have consistency. 

`inputAmount = _getBalance(token);`